core/src/main/java/org/wildfly/channel/proxy/HttpProxy.java

```diff
@@ -0,0 +1,54 @@
+package org.wildfly.channel.proxy;
+
+import org.apache.http.HttpHost;
+import org.jboss.logging.Logger;
+
+import java.net.URL;
+
+public abstract class HttpProxy {
+
+    private static final Logger LOG = Logger.getLogger(HttpProxy.class);
+
+    private HttpProxy() {
+    }
+
+    public static String PROXY_AUTHORIZATION = "Proxy-Authorization";
+
+    public static String HTTP_PROXY_HOST = "http.proxyHost";
+    public static String HTTP_PROXY_PORT = "http.proxyPort";
+    public static String HTTP_PROXY_USERNAME = "http.proxyUser";
+    public static String HTTP_PROXY_PASSWORD = "http.proxyPassword";
+
+    public static String HTTP_PROXY_PROTOCOL = "http.proxyProtocol";
+
+
+    public static String HTTPS_PROXY_HOST = "https.proxyHost";
+    public static String HTTPS_PROXY_PORT = "https.proxyPort";
+    public static String HTTPS_PROXY_USERNAME = "https.proxyUser";
+    public static String HTTPS_PROXY_PASSWORD = "https.proxyPassword";
+
+    public static String HTTPS_PROXY_PROTOCOL = "https.proxyProtocol";
+
+    public static void setup() {
```

Comment:
afaict, this method is the only one that need to be public.

The rest of this class can be package-private instead right?